### PR TITLE
Migrated popup and marker

### PR DESCRIPTION
**Pull request by HarelM**
Migrated Popup and Marker and changed the Map group to "Main":
It seems that setting the `## Events` section below the examples helps with the grouping for some reason...
image

### CODE REVIEW

**Comment by birkskyum** [comment]
It seems like that if the only comment to a return tag is `this`, then it's actually cleaner to not have it because it's reflected from the code, which is impressive because this specific addTo function from hash.ts doesn't even state `: Hash`.

Screenshot 2023-07-03 at 23 27 45
Screenshot 2023-07-03 at 23 27 22

File: src/ui/marker.ts [hunk 1]
```diff
@@ -12,39 +12,74 @@ import type {LngLatLike} from '../geo/lng_lat';
 import type {MapMouseEvent, MapTouchEvent} from './events';
 import type {PointLike} from './camera';
 
+type Alignment = 'map' | 'viewport' | 'auto';
```
Comment (birkskyum) [inline src/ui/marker.ts:15]: TSDoc comment above this will include it in the docs

File: src/ui/marker.ts [hunk 2]
```diff
@@ -12,39 +12,74 @@ import type {LngLatLike} from '../geo/lng_lat';
...
+
 type MarkerOptions = {
```
Comment (birkskyum) [inline src/ui/marker.ts:17]: TSDoc comment above this will include it in the docs

**Comment by birkskyum** [comment]
It seems like the already migrated NavigationControl also lack a TSDoc comment above the NavigationOptions.

File: src/ui/marker.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/marker.ts:15]: I wasn't sure what to write here honestly...

**Comment by HarelM** [comment]
Thanks for the review!

File: src/ui/marker.ts (hunk 1 above)
Comment (birkskyum) [reply src/ui/marker.ts:18]: Just the name of the type is sufficient to make typedoc included it

Comment (HarelM) [reply src/ui/marker.ts:18]: I added something...

File: src/ui/marker.ts (hunk 2 above)
Comment (birkskyum) [reply src/ui/marker.ts:20]: I unresolved this, because it's still missing

Screenshot 2023-07-04 at 00 02 20

Comment (HarelM) [reply src/ui/marker.ts:20]: ahh, my bad, sorry... I thought it was a duplicate comment.

Comment (birkskyum) [reply src/ui/marker.ts:20]: I've found that even adding just the following will include the respective class/methods/function/type , so if it's not always clear what to write, that is a fast improvement

```
/**
 */
 ```

**Comment by HarelM** [comment]
@birkskyum note that I removed a non-documented part of the `Marker` constructor.
I'm not sure it qualifies for a major version release, and if so I'll revert it, but this basically is a non documented behavior, I've mainly aligned the code with the docs.
Let me know what you think.
Other comments were addressed.

**Review by birkskyum — APPROVED** [review]

File: src/ui/map.ts
Comment (birkskyum) [inline src/ui/map.ts:1]: Replace `devalueValue` with `defaultValue`

**Comment by birkskyum** [comment]
> @birkskyum note that I removed a non-documented part of the `Marker` constructor. I'm not sure it qualifies for a major version release, and if so I'll revert it, but this basically is a non documented behavior, I've mainly aligned the code with the docs. Let me know what you think. Other comments were addressed.

I see, the legacyOptions - I think it's reasonable to remove it like this.

**Comment by HarelM** [comment]
I really like how less verbose and in the right place the new comments are, instead of lay around in places that are not relevant.
Merging as CI will run on the draft PR.

**Pull request merged**